packages/nuxt/src/core/app.ts

```diff
@@ -119,7 +120,7 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
 
   // Resolve middleware/ from all config layers, layers first
   app.middleware = []
-  for (const config of nuxt.options._layers.map(layer => layer.config).reverse()) {
+  for (const config of layerConfigs.reverse()) {
```

Comment:
This will mutate `layerConfigs` meaning it will have side effects which is why the tests are failing.

Reply:
But would the original options._layers not get mutated as well in that case? I mean, the logic that is applied is the same..

Reply:
What if we use slice()? Or is the only way of making it not have side effects actually calling map each time?

Reply:
**I know what can be done**